crates/uv/src/commands/toolchain/install.rs

```diff
@@ -27,85 +29,129 @@ pub(crate) async fn install(
+                toolchain.key()
+            )?;
+            if force {
+                unfilled_requests.push(request);
+            }
+        } else {
+            unfilled_requests.push(request);
+        }
+    }
 
-        if force {
-            writeln!(printer.stderr(), "Forcing reinstallation...")?;
+    if unfilled_requests.is_empty() {
+        if targets.is_empty() {
+            writeln!(
+                printer.stderr(),
+                "A toolchain is already installed. Use `uv toolchain install <request>` to install a specific toolchain.",
+            )?;
+        } else if targets.len() > 1 {
+            writeln!(
+                printer.stderr(),
+                "All requested toolchains already installed."
+            )?;
         } else {
-            if matches!(request, ToolchainRequest::Any) {
-                writeln!(
-                    printer.stderr(),
-                    "A toolchain is already installed. Use `uv toolchain install <request>` to install a specific toolchain.",
-                )?;
-            } else {
-                writeln!(
-                    printer.stderr(),
-                    "Already installed at {}",
-                    toolchain.path().user_display()
-                )?;
-            }
-            return Ok(ExitStatus::Success);
+            writeln!(printer.stderr(), "Requested toolchain already installed.")?;
         }
+        return Ok(ExitStatus::Success);
     }
 
-    // Fill platform information missing from the request
-    let request = PythonDownloadRequest::from_request(request)?.fill()?;
+    let s = if unfilled_requests.len() == 1 {
+        ""
+    } else {
+        "s"
+    };
+    writeln!(
+        printer.stderr(),
+        "Installing {} toolchain{s}",
+        unfilled_requests.len()
+    )?;
 
-    // Find the corresponding download
-    let download = PythonDownload::from_request(&request)?;
-    let version = download.python_version();
+    let downloads = unfilled_requests
+        .into_iter()
+        .map(PythonDownloadRequest::fill)
+        .collect::<Result<Vec<_>, uv_toolchain::downloads::Error>>()?
+        .iter()
+        .map(PythonDownload::from_request)
+        .collect::<Result<Vec<_>, uv_toolchain::downloads::Error>>()?;
 
     // Construct a client
     let client = uv_client::BaseClientBuilder::new()
         .connectivity(connectivity)
         .native_tls(native_tls)
         .build();
 
-    writeln!(printer.stderr(), "Downloading {}", download.key())?;
-    let result = download.fetch(&client, toolchain_dir).await?;
+    let mut tasks = futures::stream::iter(downloads.iter())
+        .map(|download| {
+            async {
+                let _ = writeln!(printer.stderr(), "Downloading {}", download.key());
+                let result = download.fetch(&client, toolchain_dir).await;
+                (download.python_version(), result)
+            }
+            .instrument(info_span!("download", key = %download))
```

Comment:
Should we instrument `PythonDownload::fetch` instead?

Reply:
I'm down for that, this was just copied from `cargo dev fetch-python`